### PR TITLE
Rails 6.1 upgrade

### PR DESCRIPTION
**Pull request by jmartin-tech**
Upgrade to Rails 6.1.

Brings the framework environment up to the current version of Rails.

## Verification

* Start `msfconsole` with various database states
- [ ] no database
- [ ] database configured via `msfdb init`
- [ ] database config in `~/.msf4/database.yml` based on `config/database.yml.example`
- [ ] database config in `config/database.yml` based on `config/database.yml.example`

* **Verify** database commands function as expected
- [ ] `hosts`
- [ ] `vulns`
- [ ] `services`
- [ ] `db_import`
- [ ] `db_nmap`

* **Verify** modules function as expected
- [ ] tester selected module 1
- [ ] tester selected module 2
- [ ] tester selected module 3
- [ ] tester selected module 4

* **Verify** sessions interactions
- [ ] windows meterpreter session (created with psexec)
- [ ] windows shell session (created with psexec)
- [ ] linux meterpreter session (created with `msfvenom`)

### CODE REVIEW

File: msfconsole [hunk 1]
```diff
@@ -13,6 +13,7 @@ begin
     Warning[:deprecated] = false
   end
 
+  $LOAD_PATH.unshift(Pathname.new(__FILE__).realpath.expand_path.parent)
```
Comment (adfoster-r7) [inline msfconsole:16]: Is it worth double checking if we'll need additional lines in the other script entrypoints? 👀

File: metasploit-framework.gemspec [hunk 2]
```diff
@@ -28,7 +28,7 @@ Gem::Specification.new do |spec|
   # only do a git ls-files if the .git folder exists and we have a git binary in PATH
   if File.directory?(File.join(File.dirname(__FILE__), ".git")) && Msf::Util::Helper.which("git")
     spec.files         = `git ls-files`.split($/).reject { |file|
-      file =~ /^documentation|^external/
+      file =~ /^external/
```
Comment (adfoster-r7) [inline metasploit-framework.gemspec:31]: Not a blocker, but I think I'm in favor of moving the swagger ruby files into the lib folder / replacing `load`s with `require`s etc

Comment (jmartin-r7) [reply metasploit-framework.gemspec:31]: Thanks for the feedback on that, checkout the updated revisions to see if you agree on the approach and namespacing I used to do this.

File: lib/msf/core/web_services/servlet/api_docs_servlet.rb [hunk 3]
```diff
@@ -45,27 +23,27 @@ def self.registered(app)
...
+          Msf::WebServices::Documentation::Api::V1::NmapApiDoc,
+          Msf::WebServices::Documentation::Api::V1::NoteApiDoc,
+          Msf::WebServices::Documentation::Api::V1::PayloadApiDoc,
+          Msf::WebServices::Documentation::Api::V1::ServiceApiDoc,
+          Msf::WebServices::Documentation::Api::V1::SessionApiDoc,
+          Msf::WebServices::Documentation::Api::V1::SessionEventApiDoc,
+          Msf::WebServices::Documentation::Api::V1::UserApiDoc,
+          Msf::WebServices::Documentation::Api::V1::VulnApiDoc,
+          Msf::WebServices::Documentation::Api::V1::VulnAttemptApiDoc,
+          Msf::WebServices::Documentation::Api::V1::WorkspaceApiDoc
```
Comment (adfoster-r7) [inline lib/msf/core/web_services/servlet/api_docs_servlet.rb:46]: 💯

**Comment by adfoster-r7** [comment]
@jmartin-r7 Looks like this needs a rebase :+1:

**Comment by adfoster-r7** [comment]
## Release Notes

Updated Metasploit's dependency on Rails from version 5.2 to 6.1

**Pull request merged**